### PR TITLE
pr297-merge revised.

### PR DESCRIPTION
**Pull request by MattOttawa**

### CODE REVIEW

File: Sources/CvPlayerAI.cpp [hunk 1]
```diff
@@ -31373,61 +31274,73 @@ CvCity* CvPlayerAI::getTeamInquisitionRevoltCity(CvUnit *pUnit, bool bNoUnit, in
 CvCity* CvPlayerAI::getReligiousVictoryTarget(CvUnit *pUnit, bool bNoUnit)
 {
 	FAssert(pUnit != NULL);
-
-	if(!hasInquisitionTarget() || (!isPushReligiousVictory() && !isConsiderReligiousVictory()))
```
Comment (billw2012) [inline Sources/CvPlayerAI.cpp]: You undid a bunch of changes here, why?

File: Sources/CvSelectionGroupAI.cpp [hunk 2]
```diff
@@ -757,10 +709,13 @@ CvUnit* CvSelectionGroupAI::AI_getBestGroupAttacker(const CvPlot* pPlot, bool bP
...
 		// if the battle occurs.  This will return a better odds estimate than we
 		//	had previously, but should not change the unit choice
-		iBestOdds = pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender);
+		iUnitOdds = pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender);
 	}
-	
-	iUnitOdds = iBestOdds;
+	else
+	{
+		iUnitOdds = iBestValue;
```
Comment (billw2012) [inline Sources/CvSelectionGroupAI.cpp]: Odds aren't the same as the value before the changes, are you sure this is correct?

Comment (MattOttawa) [reply Sources/CvSelectionGroupAI.cpp]: This bit is rather odd actually.  iUnitOdds doesn't seem to be used. If pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender) does nothing exceptreturn something, the whole part can be removed.

File: Sources/CvSelectionGroupAI.cpp [hunk 3]
```diff
@@ -669,67 +646,43 @@ CvUnit* CvSelectionGroupAI::AI_getBestGroupAttacker(const CvPlot* pPlot, bool bP
...
 			{
-				if (bForce || pLoopUnit->canMove())
+				if ((!pLoopUnit->AI_getHasAttacked() || bSuprise) && (bForce || pLoopUnit->canMove()))
 				{
 					CvUnit* pBestDefender = NULL;
 					if (bForce || pLoopUnit->canMoveInto(pPlot, moveCheckFlags, &pBestDefender))
 					{
 						PROFILE("AI_getBestGroupAttacker.RegularAttackOdds");
 
-						const int iOdds = pBestDefender?
```
Comment (billw2012) [inline Sources/CvSelectionGroupAI.cpp]: I'm talking about here: `iOdds` is the value returned by `AI_attack*` functions, `iValue` is the same but further modified below. That makes `iBestOdds` and `iBestValue` different, so you can't just remove iBestOdds. Did I miss something here?

**Comment by billw2012** [comment]
Why did you close it? There was only one more question remaining to answer!

**Comment by MattOttawa** [comment]
9 conflicts after commit to revert that.  I'll just get rid of more files.

**Comment by raxo2222** [comment]
What you tried to do?
It was meant to remove spawning of barbarian cities outside Earth mapcategory tiles - currently hills allow barbarians to spawn in space

**Comment by MattOttawa** [comment]
I'll get that in the mod but it's turned off... actually thinking now, I'll make an xml global define...

**Comment by raxo2222** [comment]
I mean 42 files were changed a lot

**Comment by MattOttawa** [comment]
haha ya, this pr got way too big, was a mistake.

**Comment by billw2012** [comment]
It was SO close! Could have accepted it and then fixed it myself.
Also you don't need to recreate your fork every time! What is that about?
Also PLEASE join discord, we can talk about this then I would have said to just wait and I will merge it myself instead of you having to redo it.

**Comment by MattOttawa** [comment]
haha sorry man, I don't know how to use github.  I'm pretty sure i still have it and I will fix it all.

**Comment by billw2012** [comment]
Wait, you are going to redo the entire thing from scratch? Or you saved it some how? Please tell me you saved it somehow!

**Pull request closed without merging**